src/internal-context/load.ts

```diff
@@ -78,6 +78,7 @@ const createInternalContext = async (moduleOptions: ModuleOptions, nuxt = useNux
         content: {
           files: [
             withSrcDir(`components/**/*${sfcExtensions}`),
+            withSrcDir(`pages/**/*${sfcExtensions}`),
```

Comment:
thanks for PR! I'm happy to approve, just wondering what happened with L93 with v3.16

Reply:
At a quick glance, I see this change around the pages config in 3.16:
https://github.com/nuxt/nuxt/pull/31090

Maybe the typing around that config property changed?

Reply:
Yes, thank you - I noticed this PR but we're checking for `options.dir.pages` which shouldn't change otherwise it would have been breaking change from the core..

Reply:
Oh wait - I think I see that `options.pages` is being changed for either being `true` or `undefined` while now an object is allowed. I believe we make appropriate changes there!

Reply:
it could be:

```ts
...(nuxtOptions.pages !== false && nuxtOptions.pages?.enabled !== false) ? [withSrcDir(`${nuxtOptions.dir?.pages || 'pages'}/**/*${sfcExtensions}`)] : []),
```

Reply:
Thanks Daniel! Yep, I was thinking this but I was also wondering if we leave the scanning to Nuxt that'll respect the patterns and we plug into `pages:extend` as well. Could be more narrow, but also improve HMR to pages dir

Reply:
nice idea.

or you could add it in any case - should be safe